pkg/storaged/client.js

```diff
@@ -969,42 +1074,56 @@ client.stratis_start = () => {
                     return true;
                 });
             });
-};
+}
 
 function stratis_fetch_properties(proxy, props) {
     const stratis = client.stratis_manager.client;
 
-    stratis.call(proxy.path, "org.storage.stratis2.FetchProperties.r4", "GetProperties", [props])
+    return stratis.call(proxy.path, "org.storage.stratis2.FetchProperties.r4", "GetProperties", [props])
             .then(([result]) => {
-                let changed = false;
                 const values = { };
                 for (const p of props) {
-                    if (result[p] && result[p][0]) {
-                        const val = result[p][1].v;
-                        if (val != proxy.data[p]) {
-                            changed = true;
-                            values[p] = val;
-                        }
-                    }
+                    if (result[p] && result[p][0])
+                        values[p] = result[p][1].v;
                 }
-                // Be careful not to add properties to a proxy that
-                // doesn't exist anymore.  That would bring it back.
-                if (changed && (proxy == client.stratis_manager ||
-                                client.stratis_pools[proxy.path] ||
-                                client.stratis_filesystems[proxy.path]))
```

Comment:
The logic here became significantly simpler, and the commit log doesn't explain why this "be careful.." is not required any more. I mostly just want to make sure that you did not forget to put this back in while working on stratis3 support?

Reply:
It is not forgotten, and thanks for bringing this up.  We needed to be careful because we were using dbus.notify.  That function will create a new proxy when it gets properties for a path that isn't in the cache (as it should).

The new code just writes to the JavaScript objects directly and triggers the client update process directly. If we do that with a object that is not longer in the cache, it's a pointless activity, but it wont add it back to the cache.

I'll add this to the commit message.

Reply:
Done.